stairway/src/main/java/bio/terra/stairway/impl/FlightContextImpl.java

```diff
@@ -89,6 +92,10 @@ public class FlightContextImpl implements FlightContext {
   // Persisted state that changes each time a step is completed and its log record is written.
   private final FlightContextLogState logState;
 
+  // -- persisted state --
+  // Progress Meters
+  private final ProgressMetersImpl progressMeters;
```

Comment:
Why is `ProgressMeterImpl` needed here (vs `ProgressReader`)? It looks like the only reason is to expose `setProgressMeter()`, but this could also be part of the `ProgressReader` interface as it doesn't expose an API a caller wouldn't already have access to via `FlightContext`.

Reply:
The code reading the FlightState does not have access to the FlightContext and should not be able to write it.

Reply:
Gotcha, that's why it's Reader and not Writer :)